### DISCUSSION TITLE
[BUG] Files within artifact zips downloaded from the web interface have the wrong modified timestamp

### DISCUSSION THREAD

**Discussion by oblivioncth** [community/community#42615] [Actions]
### Select Topic Area

Bug

### Body

Something with the way that GitHub handles artifacts is causing the modified time of files to be set incorrectly.

If you upload and download an artifact on a runner during the same job you can see that the timestamps of all files within the artifact get changed to be relative to when the download was initiated. Since the runner is obviously on the same time zone in both steps there ends up being no noticeable issue with this other than the original times not being preserved.

Before:
```
Name                : QxCoreConfig.cmake
Exists              : True
CreationTime        : 12/25/2022 5:54:25 PM
CreationTimeUtc     : 12/25/2022 5:54:25 PM
LastAccessTime      : 12/25/2022 5:54:25 PM
LastAccessTimeUtc   : 12/25/2022 5:54:25 PM
LastWriteTime       : 12/25/2022 5:54:25 PM
LastWriteTimeUtc    : 12/25/2022 5:54:25 PM
```

After
```
Name                : QxCoreConfig.cmake
Exists              : True
CreationTime        : 12/25/2022 5:58:56 PM
CreationTimeUtc     : 12/25/2022 5:58:56 PM
LastAccessTime      : 12/25/2022 5:58:56 PM
LastAccessTimeUtc   : 12/25/2022 5:58:56 PM
LastWriteTime       : 12/25/2022 5:58:56 PM
LastWriteTimeUtc    : 12/25/2022 5:58:56 PM
```

However, if I then go and download that same artifact through the web interface the modified date of the file weirdly gets changed so that whatever the actual timestamp of the file should be in UTC/GMT is set as the time in my local time zone, EST.

Example - During workflow run (UTC):
```
Name                : QxCoreConfig.cmake
Exists              : True
CreationTime        : 12/25/2022 6:36:31 PM
CreationTimeUtc     : 12/25/2022 6:36:31 PM
LastAccessTime      : 12/25/2022 6:36:31 PM
LastAccessTimeUtc   : 12/25/2022 6:36:31 PM
LastWriteTime       : 12/25/2022 6:36:31 PM
LastWriteTimeUtc    : 12/25/2022 6:36:31 PM
```

Example - After downloading through web UI (EST)
```
Name                : QxCoreConfig.cmake
Exists              : True
CreationTime        : 12/25/2022 1:42:48 PM
CreationTimeUtc     : 12/25/2022 6:42:48 PM
LastAccessTime      : 12/25/2022 1:42:48 PM
LastAccessTimeUtc   : 12/25/2022 6:42:48 PM
LastWriteTime       : 12/25/2022 6:39:36 PM
LastWriteTimeUtc    : 12/25/2022 11:39:36 PM
```

There is a 3 minute discrepancy here because I had to wait for the rest of the workflow to finish, but as you can see, the local modified file time (LastWriteTime) gets set so that its what the modified time should be in UTC, 6:39:36 PM. This leads to the actual UTC modified time being 5 hours ahead, 11:39:36 PM, which is the difference between UTC and EST. It's like whatever system is handling the files is doing the opposite of what it should be when trying to compensate for time zones.

To highlight the issue...

On runner:
```
LastWriteTime       : 12/25/2022 6:36:31 PM [UTC]
LastWriteTimeUtc    : 12/25/2022 6:36:31 PM [UTC]
```

Local system (3 minutes later):
```
LastWriteTime       : 12/25/2022 6:39:36 PM [EST]
LastWriteTimeUtc    : 12/25/2022 11:39:36 PM [UTC]
```

This is an issue because if I want to use these files locally I have to painstakingly manually change their modified dates or else CMake completely breaks down due to the dates being in the future.

**Comment by tenzap** [comment]
https://github.com/actions/upload-artifact/issues/384 ?
